regression/regdb.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+
+from testing_tool import (TestCase, apply_transform_over_tests, FAIL_MODES)
+
+import sys
+import os.path
+import getopt
+import shlex
+
+SOLVERS = [
+    'smtlib',
+    'z3',
+    'boolector',
+    'cvc',
+    'mathsat',
+    'yices',
+    'bitwuzla'
+]
+
+EXTENSIONS = {
+    'c'     : {'c', 'i'},
+    'cxx'   : {'cc', 'cpp'},
+    'cuda'  : {'cuda'},
+    'py'    : {'py'},
+    'sol'   : {'solast'},
+    'jimple': {'jimple'},
+}
+
+OPT2FLAGS = {
+    '--' + solver: {solver} for solver in SOLVERS
+} | { # Frontend related
+    '--16'           : {'16'},
+    '--32'           : {'32'},
+    '--64'           : {'64'},
+    '--binary'       : {'goto'},
+    '--little-endian': {'le'},
+    '--big-endian'   : {'be'},
+    '--no-arch'      : {'ne', 'no-arch'},
```

Comment:
I am not familiar with low lever stuff, so I don't know how common it is. But I don't think "be", "le" and "ne" are intuitive.

Reply:
"be" and "le" are quite common abbreviations, e.g., for the target triples "armel-unknown-linux-gnu". The point was for these flags to be brief, so they are easy to type and read, in particular when there are many of them. The options `-l -v` give a list of all known flags together with descriptions. The descriptions are still missing, unfortunately.

Reply:
It's fine then. Just to be sure that it's a common thing.